lntest/node.go

```diff
@@ -104,6 +109,15 @@ func nextAvailablePort() int {
 	panic("no ports available for listening")
 }
 
+// ApplyPortOffset reads the passed -portoffset flag and applies it to the next
+// available port.
+func ApplyPortOffset() {
```

Comment:
We could use the less error prone `getFreePort()` function from `channeldb/kvdb/etcd/embed.go` in which case we won't need the `lastPort` and `portOffset`.

Reply:
I started using that but now I'm not sure if it actually is better. Because it does a `listener.Close()` just before returning the port number, that port now essentially is free again. And in a system where things happen in parallel, this could lead to conflicts. I'm going to revert back to the `lastPort` variable but make the offset more dynamic.

Reply:
One way to help mitigate this is that we could use port ranges where slices try and look for open ports.

Reply:
Isn't that exactly what the `nextAvailablePort()` in combination with the offset does?